enzy_htp/mutation/api.py

```diff
@@ -406,6 +410,102 @@ def mutate_stru_with_pymol(
     return stru_cpy
 
 
+def mutate_stru_with_rosetta(stru: Structure, mutant: List[Mutation], in_place: bool = False, **kwargs) -> Union[Structure, None]:
+    """Applies mutations to a function using the Rosetta engine. 
+
+    Args:
+        stru: the 'WT' structure
+        mutant: a list of Mutation() which describes a mutant to the 'WT'
+        in_place: if make the changes to the structure in-place 
+    Returns:
+        The mutated structure. May be nothing.
+    """
+
+    # san check
+    elements: List[Dict] = list()
+    for midx, mut in enumerate(mutant):
+        if not isinstance(mut, Mutation):
+            _LOGGER.error(f"mutant takes only a list of Mutation(). Current mutant is: {mutant}")
+            raise TypeError
+
+        target = mut.target
+        if len(target) != 3:
+            target = chem.convert_to_canonical_three_letter(target)
+
+        #TODO(CJ): remap the residue index here
+        elements.extend([
+            {
+                'parent': 'RESIDUE_SELECTORS',
+                'tag': 'Index',
+                'name': f"residue{midx+1}",
+                'resnums': str(mut.res_idx)
+            },
+            {
+                'parent': 'MOVERS',
+                'tag': 'MutateResidue',
+                'name': f'mutate{midx+1}',
+                'residue_selector': f"residue{midx+1}",
+                'new_res': target
+            },
+            {
+                'parent': 'PROTOCOLS',
+                'tag': 'Add',
+                'mover_name': f'mutate{midx+1}'
+            },
+        ])
+
+    #stru_cpy = copy.deepcopy(stru)
+
+    if_retain_order = True
+    #TODO(CJ): need to remap the residue names at some point
+    #TODO(CJ): use the scratch_dir
+    parser = PDBParser()
+    stru_content: str = parser.get_file_str(stru)
+    temp_file: str = '__mut_temp.pdb'
+    fs.write_lines(temp_file, stru_content.splitlines())
```

Comment:
We can wrap this part as a method in RosettaInterface that convert Structure() -> path_of_rosetta_happy_pdb. As far as I know Rosetta dont like names like HID HIP etc. and need a {rosetta_home}/tools/protein_tools/scripts/clean_pdb.py process in general according to Rocco.